Be clearer about protections (#3900)
quicwg/base-drafts: draft-ietf-quic-transport.md

connection, or if the removal of packet protection is unsuccessful once the
expected keys are available.

Invalid packets that lack strong integrity protection, such as Initial, Retry,
or Version Negotiation, MAY be discarded. An endpoint MUST generate a

> connection error if processing of the contents of these packets prior to

Reviewer: ```suggestion
connection error if processing the contents of these packets prior to
```

> connection error if processing of the contents of these packets prior to

Reviewer: Which error?

> discovering an error resulted in changes to the state of a connection that

Reviewer: ```suggestion
discovering an error resulted in changes to connection state that
```

## Protected Packets {#packet-protected}

QUIC packets have different levels of cryptographic protection based on the
type of packet. Details of packet protection are found in {{QUIC-TLS}}; this
section includes an overview of the protections that are provided.

Version Negotiation packets have no cryptographic protection; see
{{QUIC-INVARIANTS}}.

> Retry packets use an authenticated encryption with associated data (AEAD;
> {{?AEAD=RFC5116}}) function to protect against accidental modification.

Reviewer: ```suggestion
Retry packets use an authenticated encryption with associated data function
(AEAD; {{?AEAD=RFC5116}}) to protect against accidental modification.
```

> Initial packets use an AEAD with keys that are statically derived. This does

Reviewer 2: The keys are generated for each connection, so I think it is confusing to use the word "statically" here. Maybe something like:

```suggestion
Initial packets use an AEAD with keys that are derived using a value that is visible on the wire, and therefore does
```

visible on the wire, and therefore does not provide effective confidentiality

> protection. Initial protection exists to ensure that the sender of the packet is

Reviewer: A bit of a run-on sentence here. Maybe: "Initial packets use an AEAD, the keys for which are derived using a value that is visible on the wire. The AEAD for Initial packets therefore does not provide effective ..."

> on the network path. Any entity that receives the Initial packet from a client

Reviewer: ```suggestion
on the network path. Any entity that receives an Initial packet from a client
```

> can recover the keys that will allow them to read the contents of packets or
> generate packets that will be successfully authenticated.

Reviewer: ```suggestion
can recover the keys that will allow them to both read the contents of the packet
and generate Initial packets that will be successfully authenticated at either
endpoint.
```

All other packets are protected with keys derived from the cryptographic
handshake. The type of the packet from the long header or key phase from the
short header are used to identify which encryption keys are used. Packets
protected with 0-RTT and 1-RTT keys are expected to have strong confidentiality

> and integrity protection; the cryptographic handshake ensures that only the

Reviewer 2: Maybe "authentication, confidentiality, and integrity protection"? I think that would be more consistent with the use of terms thatwe see in RFC 8446 section 1.

Author: Those are inclusive of other properties provided by TLS, so the packet protection doesn't strictly provide authentication in the same way.

Reviewer 2: I'm not against viewing that way, though then I think it might be better to talk about 0-RTT, Handshake and 1-RTT packets, rather than just 0-RTT and 1-RTT packets.

It might be a good idea to change the first sentence of the paragraph to the following and remove the last sentence: _All other packets are expected to provide strong confidentiality and integrity protection using the keys derived from the cryptograph handshake; the handshake ensures that only the communicating endpoints receive the corresponding keys._

communicating endpoints receive the corresponding keys.

> The Packet Number field that appears in some packet types contains a packet
> number, which has additional confidentiality protection that is applied as part
> of header protection; see Section 5.4 of {{QUIC-TLS}} for details. The
> underlying packet number increases with each packet sent in a given packet
> number space; see {{packet-numbers}} for details.

Reviewer: ```suggestion
The Packet Number field that appears in some packet types
has additional confidentiality protection that is applied as part
of header protection; see Section 5.4 of {{QUIC-TLS}} for details. The
underlying packet number increases with each packet sent in a given packet
number space; see {{packet-numbers}} for details.
```
